dask/array/core.py

```diff
@@ -234,34 +238,35 @@ def getem(
 ):
     """Dask getting various chunks from an array-like
 
-    >>> getem('X', chunks=(2, 3), shape=(4, 6))  # doctest: +SKIP
+    >>> graph_from_arraylike('X', chunks=(2, 3), shape=(4, 6))  # doctest: +SKIP
     {('X', 0, 0): (getter, 'X', (slice(0, 2), slice(0, 3))),
      ('X', 1, 0): (getter, 'X', (slice(2, 4), slice(0, 3))),
      ('X', 1, 1): (getter, 'X', (slice(2, 4), slice(3, 6))),
      ('X', 0, 1): (getter, 'X', (slice(0, 2), slice(3, 6)))}
 
-    >>> getem('X', chunks=((2, 2), (3, 3)))  # doctest: +SKIP
+    >>> graph_from_arraylike('X', chunks=((2, 2), (3, 3)))  # doctest: +SKIP
     {('X', 0, 0): (getter, 'X', (slice(0, 2), slice(0, 3))),
      ('X', 1, 0): (getter, 'X', (slice(2, 4), slice(0, 3))),
      ('X', 1, 1): (getter, 'X', (slice(2, 4), slice(3, 6))),
      ('X', 0, 1): (getter, 'X', (slice(0, 2), slice(3, 6)))}
     """
+    # TODO: probably get shape, dtype(?) from arr
     out_name = out_name or arr
     chunks = normalize_chunks(chunks, shape, dtype=dtype)
-    keys = product([out_name], *(range(len(bds)) for bds in chunks))
-    slices = slices_from_chunks(chunks)
 
     if (
         has_keyword(getitem, "asarray")
         and has_keyword(getitem, "lock")
         and (not asarray or lock)
     ):
-        values = [(getitem, arr, x, asarray, lock) for x in slices]
+        getter = partial(getitem, arr, asarray=asarray, lock=lock)
     else:
         # Common case, drop extra parameters
-        values = [(getitem, arr, x) for x in slices]
+        getter = partial(getitem, arr)
```

Comment:
In general we prefer to avoid including `partial`, lambdas, closures, or other dynamically generated functions in task graphs.  They're harder to serialize.